components/camel-nats/src/main/java/org/apache/camel/component/nats/NatsProducer.java

```diff
@@ -104,18 +112,12 @@ public boolean process(Exchange exchange, AsyncCallback callback) {
             });
             return false;
         } else {
-            LOG.debug("Publishing to topic: {}", config.getTopic());
-
-            final NatsMessage.Builder builder = NatsMessage.builder()
-                    .data(body)
-                    .subject(config.getTopic())
-                    .headers(this.buildHeaders(exchange));
-
-            if (ObjectHelper.isNotEmpty(config.getReplySubject())) {
-                final String replySubject = config.getReplySubject();
-                builder.replyTo(replySubject);
+            LOG.debug("Publishing to subject: {}", config.getTopic());
```

Comment:
This code in this else block needs to be in a try .. catch (exception) exchange.setExchange(e) callback done(true) idiom that are used with camel routing engine.

Reply:
Sorry, I do not understand your comment. 

`else` block is the existing code. It was not inside `try .. catch (exception)`.